<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="launch_segmentation"   default="true"/>
    <arg name="pcl_topic"             default="/camera/depth_registered/points"/>
    <arg name="max_seg_cluster_size"  default="30000" />

    <arg name="seg_node_name"           default="/rail_segmentation"/>
    <arg name="kit_icp_node_name"       default="/kit_template_matcher_node"/>
    <arg name="detect_frame"          default="base_link"/>
    <arg name="viz_detections"        default="true"/>


    <!-- start the table top segmentation -->
    <group if="$(arg launch_segmentation)">
      <node pkg="rail_segmentation" type="rail_segmentation" name="$(arg seg_node_name)">
          <param name="max_cluster_size"      value="$(arg max_seg_cluster_size)"/>
          <param name="point_cloud_topic"     value="$(arg pcl_topic)"/>
      </node>
      <node pkg="rail_segmentation_tools" type="merger" name="merger" />
    </group>

    <!-- launch bin_detector -->
    <node pkg="fetchit_bin_detector" type="bin_detector_node" name="bin_detector_node" output="screen">
        <param name="segmentation_node" value="$(arg seg_node_name)"/>
        <param name="segmentation_frame" value="$(arg detect_frame)"/>
        <param name="visualize" value="$(arg viz_detections)"/>
        <param name="kit_icp_node" value="$(arg kit_icp_node_name)"/>
    </node>

</launch>
